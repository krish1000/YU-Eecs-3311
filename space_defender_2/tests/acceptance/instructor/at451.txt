--Test errors in debug mode. Ensures second state number greater than 9 works properly.
play(5,10,1,1,1,1,1)
setup_next(3)
setup_select(5)
setup_next(5)

special
toggle_debug_mode
special
special
move(F,1)
move(A,11)
special
special
move(F,1)
move(A,11)
toggle_debug_mode
