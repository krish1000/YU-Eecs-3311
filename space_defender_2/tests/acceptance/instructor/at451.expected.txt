  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->play(5,10,1,1,1,1,1)
  state:weapon setup, normal, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_next(3)
  state:power setup, normal, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_select(5)
  state:power setup, normal, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
->setup_next(5)
  state:in game(0.0), normal, ok
  Starfighter:
    [0,S]->health:70/70, energy:70/70, Regen:1/3, Armour:1, Vision:13, Move:10, Move Cost:3, location:[C,1]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
      score:0
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->special
  state:in game(0.1), normal, error
  Not enough resources to use special.
->toggle_debug_mode
  state:in game(0.2), debug, ok
  In debug mode.
->special
  state:in game(0.3), debug, error
  Not enough resources to use special.
->special
  state:in game(0.4), debug, error
  Not enough resources to use special.
->move(F,1)
  state:in game(0.5), debug, error
  Cannot move outside of board.
->move(A,11)
  state:in game(0.6), debug, error
  Cannot move outside of board.
->special
  state:in game(0.7), debug, error
  Not enough resources to use special.
->special
  state:in game(0.8), debug, error
  Not enough resources to use special.
->move(F,1)
  state:in game(0.9), debug, error
  Cannot move outside of board.
->move(A,11)
  state:in game(0.10), debug, error
  Cannot move outside of board.
->toggle_debug_mode
  state:in game(0.11), normal, ok
  Not in debug mode.
